staging/src/k8s.io/client-go/pkg/api/meta/help.go

```diff
@@ -29,9 +29,8 @@ func IsListType(obj runtime.Object) bool {
 	// if we're a runtime.Unstructured, check to see if we have an `items` key
 	// This is a list type for recognition, but other Items type methods will fail on it
 	// and give you errors.
-	if unstructured, ok := obj.(*runtime.Unstructured); ok {
-		_, ok := unstructured.Object["items"]
-		return ok
+	if unstructured, ok := obj.(runtime.Unstructured); ok {
+		return unstructured.IsList()
```

Comment:
UnstructuredList is typed - i.e. it is tied to `meta/v1` which is a particular schema of meta.  IsList decouples the details of the underlying schema - it's the generic interface that needs no details of metadata.  Basically here's how i think of this hierarchy:

```
meta/v1 - concrete types that access data
meta/v1/unstructured - concrete types that access a particular schema of metadata
meta/v1/interfaces.go#Object - interface exposing the v1 schema, not tied to concrete types
runtime/interfaces#Unstructured - interface exposing the idea of unstructured, not tied to the v1 schema
```

Does that make sense?